components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/loader/UIBasedConfigurationLoader.java

```diff
@@ -0,0 +1,280 @@
+            throw new FrameworkException("ServiceProvider cannot be null");
+        }
+        AuthenticationStep[] authenticationSteps = serviceProvider.getLocalAndOutBoundAuthenticationConfig()
+                .getAuthenticationSteps();
+
+        return getSequence(serviceProvider, tenantDomain, authenticationSteps);
+    }
+
+    public SequenceConfig getSequence(ServiceProvider serviceProvider, String tenantDomain,
+            AuthenticationStep[] authenticationSteps) throws FrameworkException {
+        if (serviceProvider == null) {
+            throw new FrameworkException("ServiceProvider cannot be null");
+        }
+        SequenceConfig sequenceConfig = new SequenceConfig();
+        sequenceConfig.setApplicationId(serviceProvider.getApplicationName());
+        sequenceConfig.setApplicationConfig(new ApplicationConfig(serviceProvider));
+
+        // setting request path authenticators
+        loadRequestPathAuthenticatos(sequenceConfig, serviceProvider);
+
+        int stepOrder = 0;
+
+        if (authenticationSteps == null) {
+            return sequenceConfig;
+        }
+
+        // for each configured step
+        for (AuthenticationStep authenticationStep : authenticationSteps) {
+
+            try {
+                stepOrder = authenticationStep.getStepOrder();
+            } catch (NumberFormatException e) {
+                stepOrder++;
+            }
+
+            // create a step configuration object
+            StepConfig stepConfig = createStepConfigurationObject(stepOrder, authenticationStep);
+
+            // loading Federated Authenticators
+            loadFederatedAuthenticators(authenticationStep, stepConfig, tenantDomain);
+
+            // load local authenticators
+            loadLocalAuthenticators(authenticationStep, stepConfig);
+
+            sequenceConfig.getStepMap().put(stepOrder, stepConfig);
+        }
+
+        return sequenceConfig;
+    }
+
+    protected StepConfig createStepConfigurationObject(int stepOrder, AuthenticationStep authenticationStep) {
+        StepConfig stepConfig = new StepConfig();
+        stepConfig.setOrder(stepOrder);
+        stepConfig.setSubjectAttributeStep(authenticationStep.isAttributeStep());
+        stepConfig.setSubjectIdentifierStep(authenticationStep.isSubjectStep());
+        return stepConfig;
+    }
+
+    protected void loadRequestPathAuthenticatos(SequenceConfig sequenceConfig, ServiceProvider serviceProvider) {
+        if (serviceProvider.getRequestPathAuthenticatorConfigs() != null
+                && serviceProvider.getRequestPathAuthenticatorConfigs().length > 0) {
+
+            List<AuthenticatorConfig> requestPathAuthenticators = new ArrayList<AuthenticatorConfig>();
+            RequestPathAuthenticatorConfig[] reqAuths = serviceProvider.getRequestPathAuthenticatorConfigs();
+
+            // for each request path authenticator
+            for (RequestPathAuthenticatorConfig reqAuth : reqAuths) {
```

Comment:
Agreed. However these are moved (re-factored) from original class UIBasedConfigurationLoader.java, hence will not touch much within to keep the history.